### PR TITLE
Align console logging example with ASP.NET Core logging example

### PR DESCRIPTION
**Pull request by reyang**
Essentially stealing good stuff from #4821.

### CODE REVIEW

File: docs/logs/getting-started-console/Program.cs [hunk 1]
```diff
@@ -17,33 +17,38 @@
 using Microsoft.Extensions.Logging;
 using OpenTelemetry.Logs;
 
-namespace SourceGeneration;
-
-public class Program
+using var loggerFactory = LoggerFactory.Create(builder =>
 {
-    public static void Main()
+    builder.AddOpenTelemetry(options =>
```
Comment (reyang) [inline docs/logs/getting-started-console/Program.cs:22]: Do we want to demonstrate resources here?

Comment (utpilla) [reply docs/logs/getting-started-console/Program.cs:22]: I think it's fine to not include `Resources` here. We could show that in customizing the SDK exmaple.

Comment (utpilla) [inline docs/logs/getting-started-console/Program.cs:22]: Rename `options` to `logging` similar to #4821? We could use this style for the other examples as well.

**Review by utpilla — APPROVED** [review]

File: docs/logs/getting-started-console/Program.cs (hunk 1 above)
Comment (vishweshbankwar) [reply docs/logs/getting-started-console/Program.cs:22]: its there -> https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/logs/customizing-the-sdk#setresourcebuilder

**Review by vishweshbankwar — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

File: docs/logs/getting-started-console/Program.cs [hunk 2]
```diff
@@ -17,33 +17,38 @@
 using Microsoft.Extensions.Logging;
 using OpenTelemetry.Logs;
 
-namespace SourceGeneration;
-
-public class Program
+using var loggerFactory = LoggerFactory.Create(builder =>
```
[the remark this answers is not in the document]
Comment (reyang) [reply docs/logs/getting-started-console/Program.cs:20]: What options do we have and which one do you like?
1. Remove `using`, leave Dispose to GC finalizer (if there is a finalizer).
2. Add a comment to tell the user.
3. Remove `using`, add an explicit `Dispose` at the end of the application.

I'll give 1) -100, 2) +1 3) +2

[the remark this answers is not in the document]
Comment (cijothomas) [reply docs/logs/getting-started-console/Program.cs:20]: A combination of 2, 3 i.e remove using, and add explicit Dispose at the end. And comments saying
1. Loggerfactory must be kept active for the logging to work. (to warn about disposing too early)
2. Loggerfactory must be disposed at the end/shutdown to make sure any in-buffers telemetry is pushed out as well. (to warn about not disposing at all)

Not something to be fixed for this PR, as this issue is there for other getting-started docs as well

**Pull request merged**